### PR TITLE
Prepare for new leaderelection.

### PR DESCRIPTION
**Pull request by mattmoor**

### CODE REVIEW

**Comment by mattmoor** [comment]
Hmm, this is going to be blocked on the same issues as: https://github.com/knative/net-kourier/pull/117

**Comment by jmprusi** [comment]
@mattmoor we just merged a PR that unpins de depends and fixes the testing issues: #122 I think we can close this one.

**Comment by mattmoor** [comment]
Well, this has other bits that we want, which change how we test leader election to be more future proof

**Comment by mattmoor** [comment]
tl;dr I'll rebase now

**Comment by mattmoor** [comment]
this should be RFAL

File: test/e2e-tests.sh [hunk 1]
```diff
@@ -32,6 +32,10 @@ done
 add_trap "for deployment in 3scale-kourier-control 3scale-kourier-gateway; do \
   kubectl -n ${KOURIER_NAMESPACE} scale deployment $deployment --replicas=1; done" SIGKILL SIGTERM SIGQUIT
 
+# Changing the bucket count and cycling the controllers will leave around stale
+# lease resources at the old sharding factor, so clean these up.
+kubectl -n ${SYSTEM_NAMESPACE} delete leases --all
```
Comment (jmprusi) [inline test/e2e-tests.sh:37]: Shouldn't this be KOURIER_NAMESPACE? or even both?

Comment (mattmoor) [reply test/e2e-tests.sh]: yep, good catch

**Pull request merged**